v10.6 Changelog (#7755)
mattermost/docs: source/about/mattermost-v10-changelog.md

- See the [changelog in progress](https://bit.ly/2nK3cVf) for details about the upcoming release.
```

(release-v10.6-feature-release)=
## Release v10.6 - [Feature Release](https://docs.mattermost.com/about/release-policy.html#release-types)

**Release day: 2025-03-14**

### Important Upgrade Notes

>  - Support for PostgreSQL v11 and v12 have been removed. The new minimum PostgreSQL version is v13+. See the :ref:`minimum supported PostgreSQL version policy <install/prepare-mattermost-database:minimum supported version policy>` documentation for details.

Reviewer: ```suggestion
 - Support for PostgreSQL v11 and v12 have been removed. The new minimum PostgreSQL version is v13+. See the [minimum supported PostgreSQL version policy](https://docs.mattermost.com/install/prepare-mattermost-database#minimum-supported-version-policy] documentation for details.
```

>  - Migration times: On a system with 12M posts, and 1M fileinfo entries, the migration takes 15s. It is also non-blocking. Note that there is no migration for MySQL installation because this optimization is only applicable for PostgreSQL.

Reviewer: ```suggestion
 - Migration times: On a system with 12M posts, and 1M fileinfo entries, the migration takes 15s. This migration is non-blocking. Note that there is no migration for MySQL deployments because this optimization is only applicable for PostgreSQL.
```

>  - Migration times: On a system with 12M posts, and 1M fileinfo entries, the migration takes 15s. This migration is non-blocking. Note that there is no migration for MySQL deployments because this optimization is only applicable for PostgreSQL.

Reviewer 2: @agnivade Could you remind me what we mean by non-blocking here? Is creating the materialized views taking this time, or is it just refreshing them asynchronously through the job?

Reviewer 3: Sorry, I should have said "non-locking". Yes, the migration takes 15s. It blocks, but does not lock read/writes to any other table.

Reviewer 2: Thanks for clarifying!

```{Important}
If you upgrade from a release earlier than v10.5, please read the other [Important Upgrade Notes](https://docs.mattermost.com/upgrade/important-upgrade-notes.html).
```

### Improvements

#### User Interface (UI)
 - Pre-packaged Boards plugin version [v9.1.1](https://github.com/mattermost/mattermost-plugin-boards/releases/tag/v9.1.1). 
 - Pre-packaged Playbooks plugin version [v2.1.1](https://github.com/mattermost/mattermost-plugin-playbooks/releases/tag/v2.1.1).
 - Pre-packaged Copilot plugin version [v1.1.1](https://github.com/mattermost/mattermost-plugin-ai/releases/tag/v1.1.1).
 - Upgraded Ukrainian language to official.

#### Administration

>  - The performance of the site statistics section in the System Console has been significantly improved for PostgreSQL installations. The ``AnalyticsSettings.MaxUsersForStatistics`` configuration setting now just disables only the ``user_counts_with_posts chart``. There are no performance issues with any other chart or statistic. We use a materialized view to refresh the statistics every day, so the post count/file count/total file count might not be perfectly accurate for the current timestamp. This is by design and intentional. We defer loading advanced statistics to a collapsed section, where users have to click it to open the line charts and plugin statistics. To keep things simple, we have removed checking the ``AnalyticsSettings.MaxUsersForStatistics`` setting for MySQL installations in places where it is not checking for PostgreSQL. This would make performance worse for MySQL. But since MySQL is anyways deprecated for new installations, we urge customers to move to PostgreSQL as soon as feasible. Migration times: On a system with 12M posts, and 1M fileinfo entries, the migration takes 15s. It is also non-blocking. Note that there is no migration for MySQL installation because this optimization is only applicable for PostgreSQL. 

Reviewer: I'd like to see us simplify this for our admin audiences:

System Console statistics now perform faster on PostgreSQL. The ``MaxUsersForStatistics`` configuration setting now only disables the **User counts with posts** chart, while all other stats remain unaffected. Post and file counts update daily, so they may not always reflect real-time data. Advanced stats, such as line charts and plugin data, are now hidden until clicked, reducing load time. No performance improvements apply to MySQL since it's scheduled for full deprecation in v11. We recommend migrating to PostgreSQL for better performance and long-term support. The migration process for PostgreSQL is quick (around 15 seconds for large systems) and non-blocking. If you're on MySQL, no migration is needed, but transitioning to PostgreSQL is strongly encouraged.

@amyblais - who should review this simplification for technical accuracy?

Reviewer 3: > The MaxUsersForStatistics configuration setting now only disables the User counts with posts chart, while all other stats remain unaffected. 

They remain unaffected because that config value is no longer needed to disable the other queries. They will always be fast now.

 - Removed the automatic Elasticsearch/OpenSearch channel index schema check and admin alerts for the same. 

>  - Migrated scheme store SELECT * queries to use structured query builder with explicit column definitions.

Reviewer 4: This is an internal implementation detail that I don't think should be surfaced in the changelog.

>  - Updated user limits to 2.5K for soft limits (admin dismissible banner) and to 5K for hard limit (admin non-dismissible banner and no additional users can be added) for unlicensed servers. 

Reviewer: ```suggestion
 - Unlicensed server limits have been updated: a soft limit of 2500 users now results in a banner notification visible by admins, and a 5K hard limit that prevents the creation or activation of users until the user count is reduced below the hard limit.
```

>  - Removed the automatic Elasticsearch/OpenSearch channel index schema check and admin alerts for the same.

Reviewer 3: Admin won't be receiving alerts in DMs saying that their elasticsearch channel index is out of date.

### Bug Fixes
 - Fixed an issue where the email address in Mattermost would not get updated if the one in SAML changed. 
 - Fixed an issue where deleted messages would still show thread replies in the channel. 
 - Fixed an error that could occur when navigating away from the threads screen. 
 - Fixed an issue where INFO level logging for ``DoActionRequest POST`` requests was missing. 
 - Fixed an issue where users did not have the ability to toggle the switcher menu in the global header using the **SPACE** and **ENTER** keys while the product branding was in focus. 
 - Fixed "An error has occurred" bar being shown with developer mode disabled. 
 - Fixed an issue where deleted threads would get stuck in the thread viewer. 
 - Fixed an issue where the channel file count was incorrect due to files not actually being submitted as part of a post. 
 - Fixed a nil pointer dereference in the Interaction Action system. 

>  - Fixed an issue with having the currently selected thread in the Unreads pane. 

Reviewer: what was fixed here? It's unclear

>  - Fixed an issue with mmctl preventing logging a harmless debug-level "error" to the logs. 

Reviewer: ```suggestion
 - Fixed an issue with mmctl preventing the logging of a harmless debug-level "error". 
```

 - Fixed an issue where the unread count in your team sidebar may be out of sync when following/unfollowing threads. 

>  - Fixed an issue with Bulk Export -- Exports will no longer stop when they encounter a missing file. 

Reviewer: ```suggestion
 - Fixed an issue with Bulk Export: Exports will no longer stop when they encounter a missing file. 
```

### config.json
New setting options were added to ``config.json``. Below is a list of the additions and their default values on install. The settings can be modified in ``config.json``, or the System Console when available.

#### Changes to all plans:
- Under ``ServiceSettings`` in ``conig.json``:

>   - Adede a new configuration setting ``EnableWebHubChannelIteration`` was added, which allows a user to control the performance of websocket broadcasting. By default, this setting is turned off. If it is turned on, it improves the websocket broadcasting performance at the expense of poor performance when users join/leave a channel. It is not recommended to turn it on unless you have atleast 200,000 concurrent users actively using Mattermost.

Reviewer: ```suggestion
  - Adede a new configuration setting ``EnableWebHubChannelIteration`` was added, which allows a user to control the performance of websocket broadcasting. By default, this setting is turned off. If it is turned on, it improves the websocket broadcasting performance at the expense of poor performance when users join/leave a channel. We don't recommended turning this on unless you have at least 200,000 concurrent users actively using Mattermost.
```

- Removed ``EnableOpenTracing`` to remove the unused ``opentracing`` support. 

### API Changes
 - Added audit logging to the ``SearchPosts`` API.
 - Added a ``metrics`` tag to ``client_perf`` endpoint.

### Open Source Components
 - Added and removed several components.

### Go Version
 - v10.6 is built with Go ``v1.22.6``.

### Known Issues

>  - Setting the license file location through an envvar still gives the option to upload a new license through the System Console, resulting in the license being overwritten by the one set through the envvar. See this [knowledge base article](https://support.mattermost.com/hc/en-us/articles/33911983851284-System-console-still-displays-old-license-after-uploading-a-new-one) on how to resolve this issue.

Reviewer: ```suggestion
 - Setting the license file location through an environment variable still gives the option to upload a new license through the System Console, resulting in the license being overwritten by the one set through the environment variable. See this [knowledge base article](https://support.mattermost.com/hc/en-us/articles/33911983851284-System-console-still-displays-old-license-after-uploading-a-new-one) on how to resolve this issue.
```
